### PR TITLE
The grinch, a new antag, for Christmas

### PR DESCRIPTION
**Pull request by ShiftyRail**
Oh fuck I fucked the maps edition
Requested by @Arthurdentist 
[gamemode] [content]

Add the Grinch, a new antag, for Christmas. He can temper with most machinery like a gremlin (he could also be made to flip tables, if you want to). He has a big bag of holding to steal your presents.
At the end of the round, the combined spessmart price of all the items in the bag will be the Grinch's score.

The Grinch can be made unable to use guns or weapons if you want to ; right now, he's more or less like a monkey.
~~Really needs spriters because right now it's literally just the default gremlin sprite and the default (human) BoH sprite~~

:cl:
 * rscadd: Adds a new antag for christmas

### CODE REVIEW

**Comment by DeityLink** [comment]
Please tell me there's a smash bros Grinch leak reference hidden in there.

**Comment by ShiftyRail** [comment]
I unfortunately do not have that reference

**Comment by DeityLink** [comment]
Have a laugh: http://gfaqssb.wikia.com/wiki/Grinch_Leak

It was a 4chan meme too.

**Comment by Arthurdentist** [comment]
Does the grinch have the ability to ventcrawl? if so, can he ventcrawl whilst carrying his backpack? I feel like he might be quite weak without some method of getting around.

**Comment by ShiftyRail** [comment]
@Arthurdentist He can kick doors around and should be able to pass under mobs, but ventcrawl can be added.

File: code/game/objects/items/weapons/storage/bluespace.dm [hunk 1]
```diff
@@ -42,6 +42,10 @@
 	if(istype(W, /obj/item/weapon/storage/backpack/holding))
 		singulocreate(W, user)
 
+/obj/item/weapon/storage/backpack/holding/grinch
+	name = "Grinch's bag"
+	desc = "He's coming to steal your presennts."
```
Comment (Arthurdentist) [inline code/game/objects/items/weapons/storage/bluespace.dm]: misspelled presents

File: code/datums/gamemode/objectives/freeform/christmas.dm [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/datum/objective/freeform/christmas
+	explanation_text = "Tis the season. And you are here to ruin it. Disrupt Christmas and steal the crew's valuables."
```
Comment (Arthurdentist) [inline code/datums/gamemode/objectives/freeform/christmas.dm:2]: this should probably be "Tis the season, and you are here to ruin it."

**Comment by strawberrytan** [comment]
I think it'd be neat if it was not very combat capable at all but could ventcrawl away to make a good cat burglar

**Comment by ShiftyRail** [comment]
Nice, an icon conflict

**Review by MadmanMartian — COMMENTED** [review]
Idea: Rather than having to edit every map, you make the grinch lair (and other antag bases in the future) a sub-map and load it like the holodeck.

**Comment by MadmanMartian** [comment]
Also, mapdiffbot's kill

**Comment by ShiftyRail** [comment]
Can this be postponed so that we can still have the Grinch this Christmas

**Pull request merged**